stats/stats/src/charts/lines/new_accounts.rs

```diff
@@ -1,114 +1,157 @@
+use std::ops::RangeInclusive;
+
 use crate::{
-    charts::{
-        cache::Cache,
-        db_interaction::{
-            chart_updaters::{ChartFullUpdater, ChartUpdater},
-            types::{DateValue, DateValueInt},
+    charts::db_interaction::types::DateValueInt,
+    data_source::{
+        kinds::{
+            data_manipulation::map::{MapParseTo, MapToString},
+            local_db::{
+                parameters::{
+                    update::batching::{
+                        parameters::{BatchMax, PassVecStep},
+                        BatchUpdate,
+                    },
+                    DefaultCreate, DefaultQueryVec,
+                },
+                LocalDbChartSource,
+            },
+            remote_db::{QueryBehaviour, RemoteDatabaseSource, StatementFromRange},
         },
+        UpdateContext,
     },
-    UpdateError,
+    missing_date::trim_out_of_range_sorted,
+    ChartProperties, Named, UpdateError,
 };
-use async_trait::async_trait;
+
 use entity::sea_orm_active_enums::ChartType;
 use sea_orm::{prelude::*, DbBackend, FromQueryResult, Statement};
-use tokio::sync::Mutex;
 
-pub struct NewAccounts {
-    cache: Mutex<Cache<Vec<DateValueInt>>>,
-}
+pub struct NewAccountsStatement;
 
-impl NewAccounts {
-    pub fn new(cache: Cache<Vec<DateValueInt>>) -> Self {
-        Self {
-            cache: Mutex::new(cache),
+impl StatementFromRange for NewAccountsStatement {
+    fn get_statement(range: Option<RangeInclusive<DateTimeUtc>>) -> Statement {
+        // we want to consider the time at range end; thus optional
+        // filter
+        if let Some(range) = range {
+            Statement::from_sql_and_values(
+                DbBackend::Postgres,
+                r#"
+                        SELECT
+                            first_tx.date as date,
+                            count(*) as value
+                        FROM (
+                            SELECT DISTINCT ON (t.from_address_hash)
```

Comment:
Cannot we use `sql_with_range_filter_opt` here?

Reply:
Because only the upper bound of the range is used in the statement. We don't want to ignore previous transactions to get correct values for "new accounts".

For example, if address A appeared both before `range.start()` and inside the range, we don't want it to be counted within our range (it was `created` only once - on the first appearance).